### DISCUSSION TITLE
Volume Control Not Working

### DISCUSSION THREAD

**Discussion by StarChild71** [MonitorControl/MonitorControl#837] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

I'm running macOS Monterey 12.0.1 and Monitor Control 4.0.2 Build 6965 and my monitor is a Dell U3415W

I can adjust the volume with the keyboard keys as well as from Monitor Control and it shows the changes in the Monitor's own controls. So it looks like it works, however the sound coming out from the monitor's speakers doesn't change. Even if I try to mute, the sound continues to play. 

Any ideas what is going on? Thanks!

### Steps to reproduce

See above. 

### Expected behavior

Monitor sound should go up/down as selected. 

### Environment Information (please complete the following information)

```markdown
- macOS version: 12.0.1
- Mac model: Mac mini
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s): Dell U3415W
- Apple Silicon/M1 (yes or no): no
```

**Comment by ryan-ph** [comment]
I'm also seeing the same issue for volume control. Brightness control works as expected.
```
- macOS version: 12.0.1
- Mac model: M1 Macbook Pro
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s): Dell U3219Q
- Apple Silicon/M1 (yes or no): Yes
```

I'm connected over USB-C to the monitor. When using m1ddc, the command is able to set/change volume settings, but the settings don't seem to affect my audio output. Happy to help debug/test changes.

**Comment by Benjemmin** [comment]
I'm seeing the same issue, display brightness works perfect however.

- macOS version: 12.0.1
- Mac model: M1 Macbook Air
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s): HP U28
- Apple Silicon/M1 (yes or no): Yes

**Comment by shhivam** [comment]
Facing the same issue.

macOS version: 12.0.1
Mac model: M1 Macbook Air
MonitorControl version: 4.0.2 Build 6965
Monitor(s): Samsung UA32H4100
Apple Silicon/M1 (yes or no): Yes

**Comment by RadRadRadRadRad** [comment]
Same issue here, brightness still works

macOS version: 12.0.1
Mac model: 2020 M1 Mac Mini
MonitorControl version: 4.0.2 Build 6965
Monitor(s): Viewsonic Elite XG270Q
Apple Silicon/M1 (yes or no): Yes

**Comment by brypwn** [comment]
Same issue here, volume not working, brightness is working. External monitor connected via HDMI.
macOS version: 12.0.1
Mac model: 2020 M1 Macbook Air
MonitorControl version: 4.0.2 Build 6965
Monitor(s): Xiaomi Mi Curved 34"
Apple Silicon/M1 (yes or no): Yes

**Reply by yonifra** [reply]
Mine also doesn't do anything in terms of volume control. But it works for Brightness. I really need that Volume to be controlled :)

**Comment by JoniVR** [comment]
To everyone who commented here, did it work in previous versions or are you new users? 

If it worked earlier, what changed on your configuration that could have caused this? (macOS version, MonitorControl version, ...)

**Comment by the0neyouseek** [comment]
If you're using an HDMI cable with an M1 mac, there's a known issue ([see this discussion](https://github.com/MonitorControl/MonitorControl/discussions/750)) and your device is using software control, not hardware, hence why you have no sound control.

**Comment by the0neyouseek** [comment]
MonitorControl does not support software volume control (only hardware).
If you want you can try some software designed for that :
- [BackgroundMusic](https://github.com/kyleneideck/BackgroundMusic) - Free and Open Source
- [SoundControl](https://staticz.com/soundcontrol/) - 25$ with trial
- [Soundsource](https://rogueamoeba.com/soundsource/) - 47$ with trial

**Reply by anthonymobile** [reply]
just tried SoundControl and its a good alternative.

**Reply by danielrotaermel** [reply]
I am using [proxy-audio-device](https://github.com/briankendall/proxy-audio-device) now. Most unobtrusive option in my opinion. Also works great for external soundcards that only have hardware volume control (e.g. old focusrite scarlett)

**Reply by zakkhoyt** [reply]
> I am using [proxy-audio-device](https://github.com/briankendall/proxy-audio-device) now. Most unobtrusive option in my opinion. Also works great for external soundcards that only have hardware volume control (e.g. old focusrite scarlett)

@danielrotaermel Thank you. `proxy-audio-device` was an easy and effective fix. 

M1Max (MBPro), LG C1 over HDMI. 

Brightness, Volume, and Mute working now.

**Reply by jamesli0318** [reply]
> I am using [proxy-audio-device](https://github.com/briankendall/proxy-audio-device) now. Most unobtrusive option in my opinion. Also works great for external soundcards that only have hardware volume control (e.g. old focusrite scarlett)

Thank you !!!
M1 MacBook Air works well.

**Comment by ryan-ph** [comment]
> If you're using an HDMI cable with an M1 mac, there's a known issue ([see this discussion](https://github.com/MonitorControl/MonitorControl/discussions/750)) and your device is using software control, not hardware, hence why you have no sound control.

Sorry, not super familiar with this, but is this the case for USB-C connections as well?

> MonitorControl does not support software volume control (only hardware).

Any reason we don't/can't add support for software control? Maybe a naive question, but seems like a natural extension since the app already supports software dimming.

**Comment by Benjemmin** [comment]
I'm using USB-C to USB-C, so I believe no HDMI protocol/connection. First time user, so I don't know if it should have worked in previous versions. I second the question from @ryan-ph, is a software implementation an option?

**Comment by waydabber** [comment]
Hi @Benjemmin - MonitorControl will not support software volume control for direct digital outputs as that is not the scope of our application. I recommend using the apps recommended by @the0neyouseek.

**Reply by Benjemmin** [reply]
Thanks for your reply @waydabber! I was wondering in what scenario the volume controls work for the Monitor Control app? Is there some way I could re-wire the monitor/speakers/laptop so that I can use your app instead of another app?

**Reply by waydabber** [reply]
Hi, the volume control in MonitorControl is intended to control the display volume via DDC. Some displays simply do not support this. If DDC brightness control works and volume control doesn'nt, then probably there is not much you can do about it. Display firmwares vary on this. :(

**Comment by baobaollf** [comment]
There are chrome extensions that can control audio volume in the browser.

**Comment by leonekwolfik** [comment]
Same issue here, volume not working, brightness is working. 

- macOS version: 12.1
- Mac model: Mini M1 8GB
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s): Iiyama G-Master GB3466WQSU
- Monitor Cable(s)/Connection(s): USB-C Thunderbolt to DisplayPort
- Apple Silicon/M1 (yes or no): yes

See discussion: [link](https://github.com/MonitorControl/MonitorControl/discussions/915)

**Comment by steve28** [comment]
I hate to add on, but I also am using the Dell U3415W.  Brightness works.  Volume control changes the monitors volume display, but the actual volume does not change UNTIL you touch the manual volume controls then it JUMPS to the setting you set with monitor control.  In other words:

1. Using the U3415W menu, set the volume to 10%
2. Start some audio playing
2. Using the Monitor Control menu, set the volume to 100%, notice the actual volume doesn't change
3. Pull up the U3415W on-screen menu and observe the menu shows the volume at 100%
4. touch one of the volume up/down controls on the U3415W - volume immediately jumps to 100% (what was set via Monitor Control)

So it does seem that the volume commands are getting through via DDC...

Using v4.0.2 Build 6965, connected with USB-C to DisplayPort cable.

**Reply by jan-kantert** [reply]
Same happens to me. For some reason the volume setting just doesnt do anything on Mac. It works on Linux and Windows for some reason. Codec related?

**Comment by baileyj333** [comment]
I'm having the same issue. The sounds options are changing through the interface. However, the sound level isn't actually changing on the monitor. It just looks like it is.

- macOS version: 12.2.1
- Mac model: Mini (2018)
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s):  LG 34WK650
- Monitor Cable(s)/Connection(s): USB-C Thunderbolt to DisplayPort
- Apple Silicon/M1 (yes or no): No

**Comment by hmjbarbosa** [comment]
Same issue here, on an older mac with Intel. The keyboard buttons change the volume levels shown on the interface, but the actual volume coming out of the monitor does not change. If I move the volume nob on the interface, the sound also does not change. 

I tried going to the volume menu like suggested by steve28, but it doesn't work either. 

macOS version: 10.14.6 (Mojave)
Mac model: MacBook Pro (2015)
MonitorControl version: 4.0.2 Build 6965
Monitor(s): HP M27ha 
Monitor Cable(s)/Connection(s): HDMI
Apple Silicon/M1 (yes or no): No

Brightness works just fine. Monitor has DDC/CI support and it is turned on.

**Reply by waydabber** [reply]
Yeah, many displays are like this, it's totally up to the display firmware - some are just badly implemented. I just got a really expensive Lenovo Thinkvision Creator Extreme and it lacks volume controls as well (DP and HDMI). DDC does not work at all over USB-C. :(

**Comment by morrisjazz** [comment]
- macOS version: 12.0.1
- Mac model: M1 Macmini
- MonitorControl version: 4.0.2 Build 6965
- Monitor(s): Samsung UA43AU8000WXZW
- Apple Silicon/M1 (yes or no): Yes

**Comment by jillkyo520** [comment]
macOS version: 12.4
Mac model: M1 Macbook pro 14“
MonitorControl version: 4.0.2
Monitor(s): Thunderobot KU27F144M
Apple Silicon/M1 (yes or no): Yes

**Comment by jan-kantert** [comment]
This also does not work for me. It seems that this is not (entirely) the fault of MonitorControl. The volume setting actually changes inside the monitor OSD menu when I change it in MonitorControl. However, the actual volume does not change for some reason. Maybe this is related to the audio codec used by Apple? Audio on the same device works fine over HDMI for me on Linux and Windows.

**Reply by waydabber** [reply]
On Linux and Windows not the display is controlled, simply the digital audio level is tuned down. Mac sends the digital signal unaltered. There are apps on Mac that can add a volume control filter but that won't control the amplifier built into the display unlike DDC controls.

**Comment by Benjemmin** [comment]
The suggestion from @danielrotaermel is great! Is there some way this can be integrated in Monitor Control? Would be great to have the settings for this driver integrated in the Monitor Control app. :)

**Comment by chaparrodaniel** [comment]
I can confirm that **MonitorControl 4.10 DOES NOT support volume control** on my MBP 2023 with M2 Max silicon and an LG 40WP95C-W 40 display.

I really like the simplicity of of MonitorControl over BetterDisplay app.

Please notify me when a new version of MonitorControl is available to try out.

OS: MacOS Ventura 13.0
MBP 2023 14 inch with M2 Max chip
RAM: 32GB
Monitor: LG 40WP95C-W 40

===

- Searched for existing issues? Yes
- Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki? Yes
- Updated MonitorControl to the latest version (if applicable)? Yes

**Comment by Smashr** [comment]
OS: MacOS Ventura 13.0
Mac Mini 2023 M2 chip
RAM: 8GB
Connection: HDMI

Would be nice to have this working as more and more devices seem to not support this.

**Comment by azgv** [comment]
Same issue. From Apple Magic Keyboard, no volume control only brightness working.

Please let me know when this is resolved

OS: MacOS Ventura 13.2.1
Mac Model: Mini Pro 2023 with M2 Pro chip
RAM: 16GB
MonitorControl version: 4.1.0
Monitor: Benq PD3220U
Connection: Thunderbolt to Thunderbolt

**Comment by frankie-wy** [comment]
- macOS version: 13.2.1
- Mac model: MacBook Pro 15-inch, 2018
- MonitorControl version: 4.1.0
- Monitor(s): BenQ EW2780Q
- Apple Silicon/M1 (yes or no): no

I'm encounter the same issue, could not control the volume.
Please help to resolve this problem, thanks!

**Reply by jcfried** [reply]
I'm seeing exactly the same problem with a Samsung 27" HDMI monitor on my Mac mini M2 using 4.1.0.  The brightness control works but the audio control doesn't.  That said, the only way to control the volume through the monitor is by using a control at the base of the monitor screen - I was hoping MonitorControl would correct this problem, but it remains.  That said I am surprised that when using YouTube videos for a sound test, I can control the volume through YouTube.com controls on the video.

**Comment by ml6156** [comment]
Same issue here. Brightness Control is working fine, but Volume Control is a no-go when the monitor connection is HDMI. The sliders move, but the volume remains the same. Volume Control works fine when connected via USB-C. Hopefully you can add this functionality to HDMI for the Apple Silicon Macs. In any case, thanks for your good work so far. :-)

-  macOS version: 13.3.1
- Mac model: M1 Mac mini
- MonitorControl version: 4.1.0 Build 7034
- Monitor(s): ViewSonic VX2485-mhu
- Apple Silicon/M2 (yes or no): Yes

**Reply by waydabber** [reply]
Guys, again - as it is mentioned in several places, **MonitorControl does not support the HDMI port of the 2018 Intel Mini, M1/M2 Minis, Studios and MBPs - nor does it support any port of M2 Pro systems currently**. Brightness control seemingly works as the app falls back to software control in the 0-50% slider range, but hardware backlight is not affected and the slider in the 50-100% range does nothing by default as DDC does not work. Volume control also won't work without DDC even if the display supports it. 

You might want to use [BetterDisplay](https://betterdisplay.pro/) as it has DDC support for all ports on all Macs + all DDC features are completely free and unlimited. There are other alternatives as well (those are paid or limited when free).

**Comment by kn666** [comment]
It's not clear to me, why brightness software fallback is in scope (thanks for that!) and volume is not. I guess if you reconsider, it'd help people who are not technical enough to install separate soft, like danielrotaermel advised https://github.com/briankendall/proxy-audio-device or similar.

**Comment by Dr-Usman** [comment]
macOS version: 15.4.1
Mac model: MacBook M2 Pro
MonitorControl version: 4.3.3
Monitor(s): Lenovo P24h-20
Apple Silicon/M2 (yes or no):  Yes

**Comment by xuemind** [comment]
macOS version: 15.5
Mac model: MacBook Air 13-inch, M4, 2025
MonitorControl version: 4.3.3 Build 7123
Monitor(s): Dell U2723QX (DDC/CI is On)
Apple Silicon/M4 (yes or no): Yes
Connected via USB-C (Thounder 4 of Macbook) to USB-C (Monitor's USC-C DP/Data/Power)

**Reply by partounian** [reply]
@xuemind did you figure out the solution to this issue?
